### PR TITLE
Synonyms - Change synonyms internal user for minimum privileges

### PR DESCRIPTION
**Pull request by carlosdelest**
Follow up to https://github.com/elastic/elasticsearch/pull/97621. After checking with the [security team](https://elastic.slack.com/archives/CA6N3HDF1/p1689328457266199), this solution can be improved to allow just the `ReloadAnalyzerAction` on all indices instead of the `manage` privilege.

### CODE REVIEW

**Review by kderusso — APPROVED** [review]
Thanks for following up on this!

**Review by slobodanadamovic — APPROVED** [review]
LGTM 🚀  

Thanks for adjusting it.

P.S. I've added team labels. The `Check labels` requires that a sub-label is applied as well, not just the team label.

**Review by mayya-sharipova — APPROVED** [review]
@carlosdelest Thanks for following up on this. Good learning for me as well.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java [hunk 1]
```diff
@@ -180,7 +181,7 @@ public class InternalUsers {
             new RoleDescriptor.IndicesPrivileges[] {
                 RoleDescriptor.IndicesPrivileges.builder().indices(".synonyms*").privileges("all").allowRestrictedIndices(true).build(),
                 // Need manage privilege to be able to reload index analyzers for all indices
```
Comment (mayya-sharipova) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java:183]: Should we remove this comment?

**Pull request merged**